cluster/nginx.go

```diff
@@ -140,18 +117,71 @@ func DeleteNginxLBDeployments(clientset *kubernetes.Clientset, deploymentName st
 		}
 		for _, deployment := range deployments.Items {
 			if deployment.Name != deploymentName {
-				// This delete option is to make sure that the pods belonging to
-				// the nginx-resolver deployment are 'collected' immediately.
 				fgPropagation := metav1.DeletePropagationForeground
 				fgDeleteOption := metav1.DeleteOptions{
 					PropagationPolicy: &fgPropagation,
 				}
-				err := extV1beta1API(clientset).Deployments("default").Delete(deployment.Name, &fgDeleteOption)
-				if err != nil {
-					close(doneCh) // the informer listening for delete deployment event needs to be stopped
-				}
+				extV1beta1API(clientset).Deployments("default").Delete(deployment.Name, &fgDeleteOption)
 			}
 		}
+		fmt.Println("Old nginx-resolver deployments deleted correctly")
+		close(doneCh)
+	}()
+	return doneCh
+}
+
+func CreateNginxResolverDeployment(clientset *kubernetes.Clientset, deploymentName string) <-chan struct{} {
+	doneCh := make(chan struct{})
+	go func() {
+		factory := informers.NewSharedInformerFactory(clientset, 0)
+		deploymentInformer := factory.Extensions().V1beta1().Deployments().Informer()
+		deploymentInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
```

Comment:
The add event doesn't mean that the pods of the deployment are in "Running" status. It is fool proof to check if the status of the 'newObj` is "Running' in UpdateFunc handler.